src/plugins/unified_search/public/filter_bar/filter_editor/lib/filter_operators.ts

```diff
@@ -83,36 +85,71 @@ export interface Operator {
    * takes precedence over {@link fieldTypes}
    */
   field?: (field: DataViewField) => boolean;
+  /**
+   * List of operators that value preserves when switching to this operator
+   */
+  getParamsFromPrevOperator?: (
+    prevOperator: Operator | undefined,
+    params: FilterMetaParams
+  ) => FilterMetaParams | undefined;
 }
 
-export const isOperator = {
+export const isOperator: Operator = {
   message: strings.getIsOperatorOptionLabel(),
   type: FILTERS.PHRASE,
   negate: false,
   id: OPERATORS.IS,
+  getParamsFromPrevOperator: (prevOperator, params) => {
+    if (!prevOperator) return;
+    if ([OPERATORS.IS, OPERATORS.IS_NOT].includes(prevOperator.id)) return params;
+    if ([OPERATORS.IS_ONE_OF, OPERATORS.IS_NOT_ONE_OF].includes(prevOperator.id)) {
+      if (Array.isArray(params) && params.length > 0) return params[0];
+    }
+  },
```

Comment:
nit: this looks like the same function as below. Perhaps we could merge them?

Reply:
Done!